app/init/websocket_manager.ts

```diff
@@ -0,0 +1,135 @@
+
+    constructor() {
+        this.previousAppState = AppState.currentState;
+    }
+
+    public init = async (serverCredentials: ServerCredential[]) => {
+        this.netConnected = Boolean((await NetInfo.fetch()).isConnected);
+        await Promise.all(
+            serverCredentials.map(
+                ({serverUrl, token}) => async () => {
+                    const {database} = DatabaseManager.serverDatabases[serverUrl];
+                    const lastDisconnect = await queryWebSocketLastDisconnected(database);
+                    try {
+                        this.createClient(serverUrl, token, lastDisconnect);
+                    } catch (error) {
+                        console.log('WebsocketManager init error', error); //eslint-disable-line no-console
+                    }
+                },
+            ),
+        );
+
+        AppState.addEventListener('change', this.onAppStateChange);
+        NetInfo.addEventListener(this.onNetStateChange);
+
+        // TODO remove event listeners?
+    }
+
+    public invalidateClient = (serverURL: string) => {
+        this.clients[serverURL]?.close();
+        delete this.clients[serverURL];
+    }
+
+    public createClient = (serverURL: string, bearerToken: string, storedLastDisconnect = 0) => {
+        const client = new WebSocketClient(serverURL, bearerToken, storedLastDisconnect);
+
+        client.setFirstConnectCallback(() => this.onFirstConnect(serverURL)); // TODO think about reconnect
+        client.setEventCallback((evt: any) => handleEvent(serverURL, evt));
+
+        //client.setMissedEventsCallback(() => {})
+        client.setReconnectCallback(() => handleReconnect(serverURL));
+        client.setCloseCallback((connectFailCount: number, lastDisconnect: number) => this.onWebsocketClose(serverURL, connectFailCount, lastDisconnect));
+
+        if (this.netConnected) {
+            client.initialize();
+        }
+        this.clients[serverURL] = client;
+
+        return this.clients[serverURL];
+    }
+
+    public closeAll = () => {
+        for (const client of Object.values(this.clients)) {
+            client.close(true);
+        }
+    }
+
+    public openAll = () => {
+        for (const client of Object.values(this.clients)) {
+            client.initialize();
+        }
+    }
+
+    public isConnected = (serverURL: string): boolean => {
+        return this.clients[serverURL]?.isConnected();
+    }
+
+    private onFirstConnect = (serverURL: string) => {
+        // TODO: Start periodic status updates
+        handleFirstConnect(serverURL);
+    }
+
+    private onWebsocketClose = (serverURL: string, connectFailCount: number, lastDisconnect: number) => {
+        if (connectFailCount <= 1) { // First fail
+            setCurrentUserStatusOffline(serverURL);
+            handleClose(serverURL, lastDisconnect);
+
+            // TODO: Stop periodic status updates
+        }
+    }
+
+    private onAppStateChange = async (appState: AppStateStatus) => {
```

Comment:
this two event handlers seems ok, but we should test it as much as possible to make sure there are not bugs here